source/index.js

```diff
@@ -57,6 +59,31 @@ module.exports = async (input = 'patch', options) => {
 	const hasLockFile = fs.existsSync(path.resolve(rootDir, 'package-lock.json')) || fs.existsSync(path.resolve(rootDir, 'npm-shrinkwrap.json'));
 	const isOnGitHub = options.repoUrl && hostedGitInfo.fromUrl(options.repoUrl).type === 'github';
 
+	let isPublished = false;
+
+	const rollback = onetime(async () => {
+		console.log('\nPublish failed. Rolling back to the previous state…');
+
+		const lastCommit = (await git.getLastCommit()).trim();
+
+		try {
+			if (lastCommit.slice(1) === util.readPkg().version && lastCommit.slice(1) !== pkg.version) { // Verify that the package's version has been bumped before deleting the last tag and commit.
```

Comment:
Move `lastCommit.slice(1)` into a `versionFromLastCommit` (better name welcome) variable for readability. Was not immediately clear to me that it was doing slice to remove the `v`.

Reply:
Maybe `lastCommitVersion` / `versionInLastCommit`?

Reply:
I opted for `versionInLastCommit`. Still not that readable, but sounds better than the other options to me.